packages/typesafe-db/src/customers.ts

```diff
@@ -122,63 +119,98 @@ function getCustomerList(
-      'c.idCardDataOverride',
-      'c.email',
-      'c.usSsn',
-      'c.usSsnOverride',
-      'c.sanctions',
-      'c.sanctionsOverride',
+    .select(({ eb, fn, val }) => [
+      'cst.id',
+      'cst.phone',
+      'cst.authorizedOverride',
+      'cst.frontCameraPath',
+      'cst.frontCameraOverride',
+      'cst.idCardPhotoPath',
+      'cst.idCardPhotoOverride',
+      selectNewestIdCardData(eb).as('idCardData'),
+      'cst.idCardDataOverride',
+      'cst.email',
+      'cst.usSsn',
+      'cst.usSsnOverride',
+      'cst.sanctions',
+      'cst.sanctionsOverride',
       'txStats.totalSpent',
       'txStats.totalTxs',
-      ref('lastTx.fiatCode').as('lastTxFiatCode'),
-      ref('lastTx.fiat').as('lastTxFiat'),
-      ref('lastTx.txClass').as('lastTxClass'),
+      'lastTx.fiatCode as lastTxFiatCode',
+      'lastTx.fiat as lastTxFiat',
+      'lastTx.txClass as lastTxClass',
       fn<Date>('GREATEST', [
-        'c.created',
+        'cst.created',
         'lastTx.created',
-        'c.phoneAt',
-        'c.emailAt',
-        'c.idCardDataAt',
-        'c.frontCameraAt',
-        'c.idCardPhotoAt',
-        'c.usSsnAt',
-        'c.lastAuthAttempt',
+        'cst.phoneAt',
+        'cst.emailAt',
+        'cst.idCardDataAt',
+        'cst.frontCameraAt',
+        'cst.idCardPhotoAt',
+        'cst.usSsnAt',
+        'cst.lastAuthAttempt',
       ]).as('lastActive'),
-      eb('c.suspendedUntil', '>', fn<Date>('NOW', [])).as('isSuspended'),
+      eb('cst.suspendedUntil', '>', fn<Date>('NOW', [])).as('isSuspended'),
       fn<number>('GREATEST', [
         val(0),
         fn<number>('date_part', [
           val('day'),
-          eb('c.suspendedUntil', '-', fn<Date>('NOW', [])),
+          eb('cst.suspendedUntil', '-', fn<Date>('NOW', [])),
         ]),
       ]).as('daysSuspended'),
     ])
-    .where('c.id', '!=', ANON_ID)
+    .where('cst.id', '!=', ANON_ID)
     .$if(options.withCustomInfoRequest, qb =>
       qb.select(({ eb, ref }) =>
         jsonArrayFrom(
           eb
             .selectFrom('customersCustomInfoRequests')
             .selectAll()
-            .where('customerId', '=', ref('c.id')),
+            .where('customerId', '=', ref('cst.id')),
         ).as('customInfoRequestData'),
       ),
     )
     .orderBy('lastActive', 'desc')
     .execute()
 }
 
-export { getCustomerList }
+function searchCustomers(searchTerm: string, limit: number = 20): Promise<any> {
+  const searchPattern = `%${searchTerm}%`
+
+  return db
+    .selectFrom(
+      db
+        .selectFrom('customers as cst')
+        .leftJoin('editedCustomerData as cstED', 'cstED.customerId', 'cst.id')
+        .select(({ eb, fn }) => [
+          'cst.id',
+          'cst.phone',
+          'cst.email',
+          sql`CONCAT(
+            COALESCE(${selectNewestIdCardData(eb)}->>'firstName', ''),
+            ' ',
+            COALESCE(${selectNewestIdCardData(eb)}->>'lastName', '')
+          )`.as('customerName'),
```

Comment:
Maybe we should stop doing this in SQL, we already have the code in the UI ("John Doe" -> "J. Doe"): https://github.com/lamassu/lamassu-server/blob/82abf24c426a76bd54d9a6742a716b178537d6c7/packages/admin-ui/src/utils/customer.js#L14-L24

I don't see a `JOIN()` or `FILTER()` or similar that could help make this correct: https://www.postgresql.org/docs/12/functions-string.html